backend/database.js

```diff
@@ -51,6 +51,52 @@ export default (sbp('sbp/selectors/register', {
       }
     })
   },
+  'backend/db/streamEntriesBefore': async function (contractID: string, before: string, limit: number): Promise<*> {
+    let currentHEAD = await sbp('chelonia/db/latestHash', contractID)
+    if (!currentHEAD) {
+      throw Boom.notFound(`contractID ${contractID} doesn't exist!`)
+    }
+
+    const hash = before || currentHEAD
+    let prefix = '['
+    if (hash === currentHEAD) {
+      const entry = await sbp('chelonia/db/getEntry', currentHEAD)
+      prefix = `["${strToB64(entry.serialize())}",`
+      limit--
+    }
```

Comment:
This section doesn't make much sense to me, and looks like a DRY violation.

`before` is not marked as being an optional parameter, and it shouldn't be an optional parameter, so this `if` statement and `const hash = before || currentHEAD` should be removed.

Reply:
If there is no `before` in API queries, `before` will be **undefined** and also `before` could be an empty string.